osm2streets-js/src/lib.rs

```diff
@@ -116,6 +130,53 @@ impl JsStreetNetwork {
     pub fn debug_movements_geojson(&self) -> String {
         self.inner.debug_movements_geojson().unwrap()
     }
+
+    // TODO I think https://github.com/cloudflare/serde-wasm-bindgen would let us just return a
+    // HashMap
+    #[wasm_bindgen(js_name = getOsmTagsForWay)]
+    pub fn get_osm_tags_for_way(&self, id: i64) -> String {
+        abstutil::to_json(&self.tags_per_way[&osm::WayID(id)])
+    }
+
+    /// Returns a GeoJSON Polygon showing a wide buffer around the way's original geometry
+    #[wasm_bindgen(js_name = getGeometryForWay)]
+    pub fn get_geometry_for_way(&self, id: i64) -> String {
+        let id = osm::WayID(id);
+
+        // The lanes, and thus width, will be the same for every road belonging to the way
+        let width = self
+            .inner
+            .roads
+            .values()
+            .find(|r| r.osm_ids.iter().any(|x| x.osm_way_id == id))
+            .map(|r| r.total_width())
+            .unwrap();
+
+        // Show a wide buffer around the way
+        let polygon = self.original_geometry_per_way[&id].make_polygons(1.5 * width);
+
+        abstutil::to_json(&polygon.to_geojson(Some(&self.inner.gps_bounds)))
+    }
+
+    /// Modifies all affected roads and only reruns `Transformation::GenerateIntersectionGeometry`.
+    #[wasm_bindgen(js_name = overwriteOsmTagsForWay)]
+    pub fn overwrite_osm_tags_for_way(&mut self, id: i64, tags: String) {
+        let id = osm::WayID(id);
+        let tags: Tags = abstutil::from_json(tags.as_bytes()).unwrap();
+
+        for road in self.inner.roads.values_mut() {
+            if road.osm_ids.iter().any(|x| x.osm_way_id == id) {
+                // TODO This could panic, for example if the user removes the highway tag
```

Comment:
Either this method should return `Result` or we can catch the panic and bubble up an error message